### PR TITLE
Help Center: adjust opening position based on last click

### PR DESCRIPTION
**Pull request by arcangelini**
## Proposed Changes

Listen and save click events and pass them to the Help Center container. Use this data to determine where the Help Center should open at.

Screen Capture on 2024-07-04 at 00-20-10

## Why are these changes being made?

* It is important for the Help Center to open relative to where it is clicked. This can be done with some simple CSS.
* This change manages the Help Center location within itself to prevent excessive CSS around WordPress.com to accommodate different Help Center button locations.
* The goal was to create a solution that doesn't require updating CSS in all the places that Help Center is being called.

## Testing Instructions

1. Open local Calypso
2. Go to `/help` and `/home` to try out the different opening locations of Help Center.
3. Try different screen orientations
4. Test in WP-ADMIN
5. Try in checkout. /checkout/YOUR_SITE.

### CODE REVIEW

**Review by arcangelini — COMMENTED** [review]
Tested in a few locations such as:
* Checkout
* WooCommerce installation
* Home
* Help

Everything seems to be opening correctly on screen even with constrained screen dimensions.

File: packages/help-center/src/hooks/use-opening-coordinates.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+		coords.transformOrigin = 'bottom';
+	} else {
+		// Align the top edge of the help center with the bottom edge of the button
+		coords.top = buttonBottomEdge + AESTHETIC_OFFSET;
+		coords.transformOrigin = 'top';
+	}
+
+	if ( buttonLeftEdge + helpCenterWidth + AESTHETIC_OFFSET > innerWidth ) {
+		// Align right edge of the help center with the right edge of the button
+		coords.left = buttonRightEdge - helpCenterWidth - AESTHETIC_OFFSET;
```
Comment (alshakero) [inline packages/help-center/src/hooks/use-opening-coordinates.ts]: Adding `AESTHETIC_OFFSET` here ruined the alignment with the button :(

It was so clean when they're aligned. I intentionally only added the gap vertically.

File: packages/help-center/src/hooks/use-opening-coordinates.ts [hunk 2]
```diff
@@ -35,14 +35,25 @@ export const calculateOpeningPosition = ( element: HTMLElement ) => {
 
 	if ( buttonLeftEdge + helpCenterWidth + AESTHETIC_OFFSET > innerWidth ) {
 		// Align right edge of the help center with the right edge of the button
-		coords.left = buttonRightEdge - helpCenterWidth;
+		coords.left = buttonRightEdge - helpCenterWidth - AESTHETIC_OFFSET;
 		coords.transformOrigin += ' right';
 	} else {
 		// Align left edge of the help center with the left edge of the button
-		coords.left = buttonLeftEdge;
+		coords.left = buttonLeftEdge + AESTHETIC_OFFSET;
```
Comment (alshakero) [inline packages/help-center/src/hooks/use-opening-coordinates.ts]: +1.

**Review by alshakero — CHANGES_REQUESTED** [review]

**Review by alshakero — APPROVED** [review]
Tested on iPhone, iPad, and desktop sizes. All look great.

**Pull request merged**